circuit_knitting/utils/equivalence.py

```diff
@@ -0,0 +1,155 @@
+# This code is a Qiskit project.
+
+# (C) Copyright IBM 2024.
+
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Equivalence utilities.
+
+.. currentmodule:: circuit_knitting.utils.equivalence
+
+.. autosummary::
+   :toctree: ../stubs/
+
+"""
+
+import numpy as np
+from qiskit.circuit import (
+    EquivalenceLibrary,
+    QuantumCircuit,
+    QuantumRegister,
+    Parameter,
+)
+from qiskit.circuit.library.standard_gates import (
+    RZGate,
+    XGate,
+    YGate,
+    ZGate,
+    HGate,
+    SGate,
+    IGate,
+    SdgGate,
+    SXGate,
+    SXdgGate,
+    TGate,
+    TdgGate,
+    RXGate,
+    RYGate,
+    PhaseGate,
+)
+
+_eagle_sel = HeronEquivalenceLibrary = EagleEquivalenceLibrary = EquivalenceLibrary()
+
+########## Single-qubit Eagle native gate set: x, sx, rz, i ##########
+# XGate
+q = QuantumRegister(1, "q")
+def_x = QuantumCircuit(q)
+def_x.append(XGate(), [0], [])
+_eagle_sel.add_equivalence(XGate(), def_x)
+
+# SXGate
+q = QuantumRegister(1, "q")
+def_sx = QuantumCircuit(q)
+def_sx.append(SXGate(), [0], [])
+_eagle_sel.add_equivalence(SXGate(), def_sx)
+
+# RZGate
+q = QuantumRegister(1, "q")
+def_rz = QuantumCircuit(q)
+theta = Parameter("theta")
+def_rz.append(RZGate(theta), [0], [])
+_eagle_sel.add_equivalence(RZGate(theta), def_rz)
+
+# IGate
+q = QuantumRegister(1, "q")
+def_i = QuantumCircuit(q)
+def_i.append(IGate(), [0], [])
+_eagle_sel.add_equivalence(IGate(), def_i)
+
+######################################################################
+
+# YGate
+q = QuantumRegister(1, "q")
+def_y = QuantumCircuit(q)
+for inst in [RZGate(np.pi), XGate()]:
+    def_y.append(inst, [0], [])
+_eagle_sel.add_equivalence(YGate(), def_y)
+
+# ZGate
+q = QuantumRegister(1, "q")
+def_z = QuantumCircuit(q)
+def_z.append(RZGate(np.pi), [0], [])
+_eagle_sel.add_equivalence(ZGate(), def_z)
+
+# HGate
+q = QuantumRegister(1, "q")
+def_h = QuantumCircuit(q)
+for inst in [RZGate(np.pi / 2), SXGate(), RZGate(np.pi / 2)]:
+    def_h.append(inst, [0], [])
+_eagle_sel.add_equivalence(HGate(), def_h)
+
+# SGate
+q = QuantumRegister(1, "q")
+def_s = QuantumCircuit(q)
+def_s.append(RZGate(np.pi / 2), [0], [])
+_eagle_sel.add_equivalence(SGate(), def_s)
+
+# SdgGate
+q = QuantumRegister(1, "q")
+def_sdg = QuantumCircuit(q)
+def_sdg.append(RZGate(-np.pi / 2), [0], [])
+_eagle_sel.add_equivalence(SdgGate(), def_sdg)
+
+# SXdgGate
+q = QuantumRegister(1, "q")
+def_sxdg = QuantumCircuit(q)
+for inst in [
+    RZGate(np.pi / 2),
```

Comment:
The `BasisTranslator` returned this. Fairly certain we can collapse to one pi rotation on either side of the sxgate